### ISSUE TITLE
System.CommandLine integration follow up issues

### ISSUE THREAD

**Issue by sfoslund** [dotnet/sdk#14461]
SDK tests that have been temporarily disabled in https://github.com/dotnet/sdk/pull/14379 as they likely require changes to System.CommandLine (cc @jonsequitur)

- **RestoreDistinguishesRepeatSourceArgsFromCommandArgs (+ 1 other)**: The previous parser required users to include an option flag before each option value (for example, "--source source1 --source source2" instead of "--source source1 source2". Since system command line does not require this, tokens that are supposed to be arguments are being matched as part of an option value. For example, parsing "--source source1 arg" should produce "arg" for the argument value but instead it is being seen as another value for "--source"
		Proposed solution: provide the ability to opt out of this multiple option values per option flag behavior -- **EDIT**: requires a system.commandline change: https://github.com/dotnet/command-line-api/issues/1077

- `-p` vs `-property` parsing issue: https://github.com/dotnet/sdk/pull/14664#discussion_r528926727

**Comment by wli3** [comment]
> RestoreCapturesArgumentsToForwardToMSBuildWhenTargetIsSpecified

do you see any down side of threating it as an argument instead of unmatched token? Do we need to "unmatched" it in order to pass it down to msbuild?

> "--source source1 --source source2" instead of "--source source1 source2"

this looks ok. double check with @KathleenDollard

**Comment by sfoslund** [comment]
> do you see any down side of threating it as an argument instead of unmatched token? Do we need to "unmatched" it in order to pass it down to msbuild?

Jon and I discussed this yesterday and came up with a plan to handle this without treating it as an unmatched token, so this should be solved. 

>>"--source source1 --source source2" instead of "--source source1 source2"

> this looks ok.

This can actually cause a lot of issues if the option is followed by an argument. For example, this behavior would break the first example in [this doc](https://docs.microsoft.com/en-us/dotnet/core/tools/global-tools-how-to-use#use-the-tool-as-a-global-tool) because anything after the --add-source option would be treated as another source, meaning the argument would be ignored. Jon and I discussed and there will likely have to be a System.CommandLine change here, we have this bug tracking: https://github.com/dotnet/command-line-api/issues/1077
